Makefile

```diff
@@ -184,3 +184,13 @@ image/push/scorecard-proxy:
 	./hack/image/push-image-tags.sh $(SCORECARD_PROXY_BASE_IMAGE):dev $(SCORECARD_PROXY_IMAGE)
 
 .PHONY: image image/scaffold/ansible image/scaffold/helm image/build image/build/ansible image/build/helm image/push image/push/ansible image/push/helm
+
+##@ Development
+
+doc/setup: ## It will install doctoc locally in order to allow you update/generate index in the docs automatically by the command $make doc/update. Ensure that you have a npm installed locally in order to use it. More info: https://www.npmjs.com/package/doctoc
+	@echo Installing npm doctoc ...
+	npm install -g doctoc
```

Comment:
Is there a `go` tool that can generate markdown table of contents? IMO we shouldn't add `npm` as a developer dependency.

Reply:
Hi @joelanford, 

Would be great have one in go that is stable, maintained and used for we adopt here since the project is in go. However, If we look in https://go-search.org/search?q=toc+markdown the most used is not really used and besides it says that works in MacOsX it did not work for me. Also, it has no contributions, is not following the good practices and shows that is not really maintained. Because of this, IMHO we cannot adopt this one and the others for similar reasons. 

However, if we check in the internet the https://www.npmjs.com/package/doctoc is one of the most used to addend this need (20.000+ downloads just this week), have maintainers, it is very easy to use, work in any SO/distro etc ...

Reply:
This one seems like it could work: https://github.com/ekalinin/github-markdown-toc (it's bash-based)

We could just vendor it into our repo like we do with `./hack/ci/marker`

I'm just really uneasy about adding an `npm` dependency and having to propagate it through to make it work in CI and ask developers to carry it.

Reply:
@hasbro17 @estroz @jmccormick2001 @theishshah @jmrodri @fabianvf Thoughts?

Reply:
Hi @joelanford, 

The project <https://github.com/ekalinin/github-markdown-toc> is that one which I spoke above. Following the problems faced with:

- the master branch is broken  
- It has a binary for each SO :-( which make very hard to use 
- Besides its docs is saying that it works in Mac Os X I could not make it work following these steps to install
- I could install using this [basher](https://github.com/basherpm/basher) which shows another project not really adopted by the community
- After all, I could make it works and it is not updating the files. It just does the output in the shell for the dev copy and past :-( 

I understand why would like to use a go tool instead of, however, IMHO the most important is adopted something that is reliable I mean which is maintained and used by the community.  The doctoc is easy to install and to use(very intuitive and all is done with 2 commands in any SO), and it is quite adopted by the community as well which means that it will be kept maintained. Also, it is just for the docs. 

Please, let me know if you find any other option for we try.

Reply:
@camilamacedo86 @joelanford there's also a go version by the same author: https://github.com/ekalinin/github-markdown-toc.go